### ISSUE TITLE
XIA:PFCU-4

### ISSUE THREAD

**Issue by Baokangwen** [epics-modules/dxp#9]
This IOC can be used for XIA's PFCU-4 controller？

**Comment by MarkRivers** [comment]
No, this is for the XIA X-ray spectroscopy pulse-processing modules, like the Saturn, Mercury, and xMAP.

The files for the PFCU-4 are in this module:
https://github.com/epics-modules/optics

This file uses the RS-232 interface to control the PFCU-4:
https://github.com/epics-modules/optics/blob/master/opticsApp/Db/XIA_shutter.db

This file uses a digital output module to control the PFCU-4.
https://github.com/epics-modules/optics/blob/master/opticsApp/Db/filterBladeNoSensor.db

**Comment by Baokangwen** [comment]
@MarkRivers 
Thank you for your reply.
I have successfully compiled this module(optics), but I don't know how to use it yet, I want to control the PF4 shutter with the PFCU-4 controller.In this module, I did not find the st.cmd file related to PFCU-4 .I want to ask for your help.

I previously downloaded a module called xia-pfcu-4-1.4.0 by J. Lewis Muir.Unfortunately, this module doesn't compile, so I don't know how to use it.(https://www.imca.aps.anl.gov/~jlmuir/index.html)

Yours sincerely,
Kangwen

**Comment by MarkRivers** [comment]
This is how I load the database for this module:
```
tyGSAsynInit("serial10", "UART1", 1, 9600,'N',1,8,'N',"\r","\r")  /* XIA shutter */
dbLoadRecords("$(OPTICS)/db/XIA_shutter.db", "P=13BMD:,S=filter1,ADDRESS=1,PORT=serial10")
```

The tyGSAsynInit command is specific to vxWorks, so you won't use that.  But it shows that you need to configure the port for 9600 baud, 1 stop bit, 8 data bits, no parity, no flow control.  You configure those with asynSetOption.

The input EOS and output EOS are both \r.  You configure those with asynOctetSetInputEos and asynOctetSetOutputEos.
